tests/test_topic_inheritance.py

```diff
@@ -11,14 +11,15 @@
 from cyclonedds.sub import DataReader
 from cyclonedds.util import duration
 
-
+BASE_IDL = 'module Hierarchy { @mutable struct Base { string fieldA; }; };'
 @dataclass
 @annotate.mutable
 @annotate.autoid("sequential")
 class Base(idl.IdlStruct, typename="Hierarchy.Base"):
     fieldA: str
 
 
+DERIVED_IDL = 'module Hierarchy { struct Derived : Base { string fieldB; }; };'
```

Comment:
```suggestion
DERIVED_IDL = 'module Hierarchy { @mutable struct Base { string fieldA; }; @mutable struct Derived : Base { string fieldB; }; };'
```

Reply:
Hi @eboasson! Thanks for fixing the "typeof" command and circling back to this pull request. Your code change suggestion is correct, but I found another minor problem in the _type_checker function. I intend to handle the code changes when I return to the office in three weeks.

Stopping supporting Python 3.7 makes sense, as it has already reached its end-of-life. I have yet to check the Python version on Ubuntu or the end-of-life dates for its versions, but it would make sense to follow what has been done in CycloneDDS core. I would need to think about that a bit further to give you a better answer.